### PR TITLE
111 implement snapshot and recording upload to gridfs

### LINKED ISSUE #111: Implement Snapshot and Recording Upload to GridFS
**Description:**
Integrate MongoDB's GridFS to store large files such as image snapshots and video recordings. Modify the `save_pic()` and `save_recording()` methods to store these files directly in MongoDB GridFS, rather than saving them locally. Store the file references in the MongoDB reports collection.

**Acceptance Criteria:**
- [ ] Video recordings are stored directly in MongoDB GridFS.
- [ ] Snapshots are stored directly in MongoDB GridFS.
- [ ] File references (e.g., file IDs) are included in the corresponding MongoDB report document.
- [ ] Images can be successfully uploaded, stored, and retrieved from GridFS.
- [ ] Videos can be successfully uploaded, stored, and retrieved from GridFS.

**Reason:**
Using GridFS ensures that large files like images and video recordings can be efficiently stored within MongoDB without size limitations. This also keeps the application scalable, avoiding the need for external file systems.

### PR DESCRIPTION
**Pull request by nguzinski**
fixes #111

allows for uploading and retrieval of images and recordings to a MongoDB

was requested for us to allow for files and tests to be stored remotely
spun up the MongoDB, (without the pass/user) spun up a gridFS, whenever images/recordings are given they are then sent into the gridFS, and can later be pulled out and reformated into the proper file.

There is some confusion about the specifics of the picture format given, and the naming scheme for the files but those can be fixed when the team comes to a consensus

### CODE REVIEW

**Review by Damerson1 — COMMENTED** [review]
This might be a placeholder, but it would be helpful to allow for dynamic file name entry rather than, "my_image.jpg"

Also, it would also be helpful to catch more specific errors such as type error, value error, etc. rather than just using an exception. 

Save_pic and save_recording use simila logic so they may be able to be combined into one singular function. 

Otherwise, the code seems to look fine.

File: backend/PythonClient/multirotor/airsim_application.py [hunk 1]
```diff
@@ -2,14 +2,18 @@
...
+from gridfs import GridFS
 from abc import abstractmethod
-
 from PythonClient import airsim
 
 
 class AirSimApplication:
     # Parent class for all airsim client side mission and monitors
     def __init__(self):
+        self.client = MongoClient("mongodb+srv://<db_username>:<db_password>@dronereports.lkmgn.mongodb.net/?retryWrites=true&w=majority&appName=DroneReports")
```
Comment (kate-holdener) [inline backend/PythonClient/multirotor/airsim_application.py:14]: It seems wrong to have this URL hard coded (even though it's parameterized with username and password). Perhaps using an environment variable would be better.

File: backend/PythonClient/multirotor/control/break_apart.py
Comment (kate-holdener) [inline backend/PythonClient/multirotor/control/break_apart.py:1]: What's this empty file for?

File: .gitignore [hunk 2]
```diff
@@ -1,9 +1,9 @@
 # Ignore Visual Studio files
 .vs/
-
+.vscode/
```
Comment (kate-holdener) [inline .gitignore:3]: I'm a little confused here. You are adding .vscode to gitignore but the repo has .vscode/settings.json in it. Does that mean that the team does want to keep track of/share .vscode settings or should .vscode directory be removed from the repo?

**Pull request closed without merging**